### DISCUSSION TITLE
How to skip sending web-hook on update via custom extension?

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#6972] [Archive]
I'm trying to skip call to webhook via custom extension like this:

```js
module.exports = function registerHook({services, exceptions, database, env}) {
    return {
        'items.*': async function(data) {
            const {event, collection, payload={}, accountability={}} = data
            if (['items.read'].includes(event)){
                return data
            }

            const {userAgent = ''} = accountability

            const _skip_webhook_call = payload._skip_webhook_call || false
            delete payload._skip_webhook_call

            if (_skip_webhook_call){
                console.log(event,collection,`Skip calling webhook`)
                return null
            }
            return data
		}
    }
}
```

but is not working, is there a way to do this? i guess having a custom exception for skipping webhook call can help.

Thx.

**Comment by rijkvanzanten** [comment]
Instead of trying to skip the native webhook, I would recommend calling the endpoint yourself in this hook instead

**Comment by botzill** [comment]
But in this case I can't use the directus webhooks management where i can
set collections, actions, enable/disable them when I need to.

**Reply by rijkvanzanten** [reply]
True, but that can easily be reproduced in the hook itself with a simple `if`:

```
if (collection === 'my-collection') {
  await axios.post('https://example.com/my-webhook', { etc });
}
```

**Comment by botzill** [comment]
yes I know, but we have many collections, user, new ones are added and we
have to edit this all the time to include the collections that we need.

Maybe we can do some flag for this to be able to do this easily?

On Mon, 26 Jul 2021 at 19:22, Rijk van Zanten ***@***.***>
wrote:

> True, but that can easily be reproduced in the hook itself with a simple
> if:
>
> if (collection === 'my-collection') {
>   await axios.post('https://example.com/my-webhook', { etc });
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/directus/directus/discussions/6972#discussioncomment-1052091>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDYBWCTWVRKUPKV222WAYLTZWDUBANCNFSM5BACWBLQ>
> .
>


-- 
Chirica Gheorghe,
Co-Founder - https://www.crawless.com

**Reply by rijkvanzanten** [reply]
> have to edit this all the time to include the collections that we need.

Sure, but doesn't the same apply to webhooks?

> Maybe we can do some flag for this to be able to do this easily?

I'm not sure if that meets the [80/20](https://docs.directus.io/contributing/introduction/#feature-requests), but lets leave this discussion open and we'll see what traction it gets from the community 🙂
